torchtext/experimental/transforms.py

```diff
@@ -174,6 +175,162 @@ def to_ivalue(self):
         return TextSequentialTransforms(OrderedDict(module_list))
 
 
+_pretrained_spm = ['text_unigram_15000', 'text_unigram_25000', 'text_unigram_50000',
+                   'text_bpe_15000', 'text_bpe_25000', 'text_bpe_50000']
+
+
+def pretrained_spm(spm_model='text_unigram_25000'):
+    r"""Generate a pretrained sentencepiece model.
+        The model was trained with torchtext.datasets.WikiText103, torchtext.datasets.EnWik9 and BookCorpus.
+        Both BPE and unigram methods were used to train the model (for more details please refer to
+        SentencePiece GitHub https://github.com/google/sentencepiece). We also provide the pretrained model
+        with a different size of the vocabulary (i.e. 15000, 25000, 50000).
+
+    Args:
+       spm_model: the pretrained sentencepiece model names. Default: 'text_unigram_25000'. The following pretrained sentencepiece models are provided:
+            - text_unigram_15000
+            - text_unigram_25000
+            - text_unigram_50000
+            - text_bpe_15000
+            - text_bpe_25000
+            - text_bpe_50000
+            Otherwise, the file path to the user-provided sentencepiece model is required.
+
+    Examples:
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+
+    """
+    if spm_model in _pretrained_spm:
+        spm_model = download_from_url('https://pytorch.s3.amazonaws.com/models/text/pretrained_spm/{}.model'.format(spm_model))
+        return load_sp_model(spm_model)
+    else:
+        raise RuntimeError('The pretrained sentencepiece model is not valid')
+
+
+class SentencePieceTokenizer(nn.Module):
```

Comment:
We have this pattern going on where there is a class wrapper for each tokenizer and vocab, mostly just to support the JIT.

```
spm = sentencepiece_tokenizer(<file-like object)
def transforms(line):
    x = spm(line)
```

Cant' do this because transforms won't be Jit-able.

So instead we've been writing these wrappers, maybe we can think of a more generic solution?

Reply:
This is the reason to have the wrapper.